llvm/include/llvm/Support/NVVMIntrinsicFlags.h

```diff
@@ -0,0 +1,39 @@
+//===--- NVVMIntrinsicFlags.h -----------------------------------*- C++ -*-===//
```

Comment:
It does not seem usual to have Target-specific intrinsic info in libSupport, are there precedents?

Reply:
I could not find exact precedents.

As I mentioned in the commit message, we would like to use these enums/flags even in upstream clients (i.e. producers of the intrinsic like NVVM Dialect for example) and not just in the NVPTX backend.  So, I tried to choose a location that looks relevant/compatible both upstream and downstream.

With that, I found a few *Metadata.h, *Attributes.h files in this directory and thought it might be an appropriate one. But, I am open to other preferences that we may have.

Reply:
Can this be in the IR directory maybe?

Reply:
IR feels appropriate, as this header contains definitions for intrinsic operand encoding.

Reply:
Sure, Thanks both of you for the direction. I will use IR directory for other intrinsics.
(For this intrinsic, we are not using flags in the latest revision.)

Resolving this,